python/packages/autogen-core/src/autogen_core/model_context/_unbounded_buffered_chat_completion_context.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any, List, Mapping
+
+from ..models import LLMMessage
+from ._chat_completion_context import ChatCompletionContext
+
+
+class UnboundedBufferedChatCompletionContext(ChatCompletionContext):
+    """An unbounded buffered chat completion context that keeps a view of the all the messages."""
+
+    def __init__(self, initial_messages: List[LLMMessage] | None = None) -> None:
+        self._messages: List[LLMMessage] = initial_messages or []
+
+    async def add_message(self, message: LLMMessage) -> None:
+        """Add a message to the memory."""
```

Comment:
While we are at it, let replace the mention of "memory" from all model context implementations. This is to avoid confusing this with the memory module.